litestar/middleware/exceptions/middleware.py

```diff
@@ -147,17 +148,30 @@ class ExceptionHandlerMiddleware:
     This used in multiple layers of Litestar.
     """
 
-    def __init__(self, app: ASGIApp, debug: bool, exception_handlers: ExceptionHandlersMap) -> None:
+    def __init__(self, app: ASGIApp, debug: bool | None, exception_handlers: ExceptionHandlersMap) -> None:
         """Initialize ``ExceptionHandlerMiddleware``.
 
         Args:
             app: The ``next`` ASGI app to call.
-            debug: Whether ``debug`` mode is enabled
+            debug: Whether ``debug`` mode is enabled. Deprecated. Debug mode will eb inferred from the request scope
```

Comment:
Wondering if we could have some of those version vars?
```rst
.. versionchanged:: 2.0.0
   The feature is deprecated.
   ```

Reply:
Sure. We could even automatically inject those with our `deprecated` decorator by manipulating the `__doc__` attribute. If you want to take a look at that.. :slightly_smiling_face: